File: lib/services/openai_service.dart
import 'dart:convert';

import 'package:http/http.dart' as http;
import 'package:raio_fibra_app_v2/core/api_config.dart';

class OpenAiService {
  static const Duration _timeout = Duration(seconds: 6);

  Future<String> enviarMensagem({
    required String mensagem,
    required List<Map<String, dynamic>> historico,
  }) async {
    try {
      final response = await http
          .post(
            Uri.parse('${ApiConfig.baseUrl}/openai/chat'),
            headers: {'Content-Type': 'application/json'},
            body: jsonEncode({'message': mensagem, 'history': historico}),
          )
          .timeout(_timeout);

      if (response.statusCode >= 200 && response.statusCode < 300) {
        final data = jsonDecode(response.body) as Map<String, dynamic>;
        final reply = data['reply'];

        if (reply is String && reply.isNotEmpty) {
          return reply;
        }
      }
    } catch (_) {}

    return _respostaLocal(mensagem);
  }

  String _respostaLocal(String mensagem) {
    final texto = mensagem.toLowerCase();

    if (texto.contains('sem internet') ||
        texto.contains('caiu') ||
        texto.contains('offline') ||
        texto.contains('sem conexao')) {
      return 'Entendi que voce esta sem internet. Verifique a luz LOS do roteador, reinicie o equipamento por 30 segundos e confira os cabos. Se continuar, posso orientar a abertura de chamado tecnico.';
    }

    if (texto.contains('lenta') ||
        texto.contains('lentidao') ||
        texto.contains('devagar') ||
        texto.contains('travando')) {
      return 'Para internet lenta, teste perto do roteador, feche downloads e videos, reinicie o equipamento e faca um teste de velocidade. Se ficar abaixo do plano, abrimos um chamado com prioridade.';
    }

    if (texto.contains('fatura') ||
        texto.contains('pix') ||
        texto.contains('boleto') ||
        texto.contains('segunda via')) {
      return 'Posso ajudar com sua fatura. A area de Faturas gera segunda via, PIX copia e cola e abre o pagamento simulado do Mercado Pago.';
    }

    if (texto.contains('chamado') || texto.contains('suporte')) {
      return 'Antes de abrir chamado, me diga o problema, quando comecou, se ja reiniciou o roteador e se a luz LOS esta vermelha. Assim o suporte recebe tudo completo.';
    }

    if (texto.contains('wifi') ||
        texto.contains('wi-fi') ||
        texto.contains('senha') ||
        texto.contains('roteador')) {
      return 'Para Wi-Fi, confira a rede, a senha, o local do roteador e a quantidade de dispositivos conectados. A area Meu Wi-Fi esta pronta para integrar esses dados.';
    }

    if (texto.contains('plano') ||
        texto.contains('upgrade') ||
        texto.contains('promocao')) {
      return 'Temos opcoes de upgrade e promocoes. Para demonstracao, posso sugerir um plano superior conforme seu perfil de uso e beneficios comerciais.';
    }

    return 'Sou a Raio IA e posso ajudar com sem internet, internet lenta, fatura, PIX, boleto, suporte, chamado, Wi-Fi, plano, upgrade e promocao. Me diga o que voce precisa.';
  }
}

File: backend/services/openai.service.js
function buildReply(message = '') {
  const text = message.toLowerCase();

  if (
    text.includes('sem internet') ||
    text.includes('caiu') ||
    text.includes('offline') ||
    text.includes('sem conexao')
  ) {
    return 'Entendi que voce esta sem internet. Verifique se a luz LOS do roteador esta vermelha, reinicie o equipamento por 30 segundos e confira os cabos. Se continuar sem conexao, posso orientar a abertura de chamado tecnico.';
  }

  if (
    text.includes('lenta') ||
    text.includes('lentidao') ||
    text.includes('devagar') ||
    text.includes('travando')
  ) {
    return 'Para internet lenta, teste perto do roteador, feche downloads e videos, reinicie o equipamento e faca um teste de velocidade. Se o resultado ficar muito abaixo do plano, abrimos um chamado com prioridade.';
  }

  if (
    text.includes('segunda via') ||
    text.includes('fatura') ||
    text.includes('pix') ||
    text.includes('boleto')
  ) {
    return 'Posso ajudar com sua fatura. Voce pode consultar a segunda via, gerar PIX copia e cola e abrir o checkout simulado do Mercado Pago pela area de Faturas ou Pagamento.';
  }

  if (
    text.includes('abrir chamado') ||
    text.includes('chamado') ||
    text.includes('suporte tecnico') ||
    text.includes('suporte')
  ) {
    return 'Antes de abrir o chamado, me diga o problema, quando comecou, se ja reiniciou o roteador e se a luz LOS esta vermelha. Com isso o suporte recebe um atendimento mais completo.';
  }

  if (
    text.includes('wifi') ||
    text.includes('wi-fi') ||
    text.includes('senha') ||
    text.includes('roteador')
  ) {
    return 'Para Wi-Fi, confira se esta na rede correta, se a senha mudou, se o roteador esta em local aberto e quantos dispositivos estao conectados. A area Meu Wi-Fi ja esta preparada para integrar esses dados.';
  }

  if (
    text.includes('plano') ||
    text.includes('upgrade') ||
    text.includes('promocao')
  ) {
    return 'Temos opcoes de upgrade e promocoes. Para demonstracao, posso sugerir sair do plano 600 Mega para um plano superior conforme perfil de uso, com beneficios comerciais e fidelizacao.';
  }

  return 'Sou a Raio IA e posso ajudar com sem internet, internet lenta, segunda via, PIX, boleto, chamado, suporte tecnico, Wi-Fi, planos, upgrade e promocoes. Me diga o que voce precisa.';
}

async function sendChatMessage({ message, history }) {
  return {
    reply: buildReply(message),
    historyLength: Array.isArray(history) ? history.length : 0,
  };
}

module.exports = {
  sendChatMessage,
};
